### PR TITLE
KILL: Add kill board package and update alarms

### PR DESCRIPTION
**Pull request by mattlangford**
Some noteworthy things changed in this PR:
- Meta alarms require none of the "sub" alarms that could raise them are raised in order for the meta alarm to be cleared. For example, if the `kill` meta-alarm is triggered by the `rc_kill` and `hw_kill` alarms, both of those alarms must be clear in order for the `kill` alarm to also be clear. This was important for the kill board node.
- Speaking of, the kill board node is here and it connected with Daniel Duggers kill board to connect it with ROS. I tested different control methods (RC and Autonomous) to make sure it lit up correctly. Also I tested different alarm set ups, everything seemed to work.
- One weird thing I did was not include `odom_kill` in the kill meta alarm because this would effectively disable RC mode if odom went down - which is an undesired behavior, so it's handled separately.

### CODE REVIEW

File: mission_control/navigator_missions/navigator_singleton/navigator.py [hunk 1]
```diff
@@ -241,10 +242,15 @@ def _make_alarms(self):
 
         fprint("Alarm listener created, listening to alarms: ", title="NAVIGATOR")
 
+        self.alarm_listener.add_listener("odom_kill", lambda alarm: setattr(self, 'odom_killed', not alarm.clear))
         self.alarm_listener.add_listener("kill", lambda alarm: setattr(self, 'killed', not alarm.clear))
+        
         yield self.alarm_listener.wait_for_alarm("kill", timeout=.5)
+        yield self.alarm_listener.wait_for_alarm("odom_kill", timeout=.5)
```
Comment (mattlangford) [inline mission_control/navigator_missions/navigator_singleton/navigator.py]: This should be odom_loss

File: mission_control/navigator_missions/nav_missions/circle_tower.py [hunk 2]
```diff
@@ -7,55 +7,114 @@
...
+
+        print TOTEM_SAFE_DIST - ROT_SAFE_DIST
+        if direction == "COUNTER-CLOCKWISE":
+            rot_move = navigator.move.yaw_right(1.57).left(TOTEM_SAFE_DIST - ROT_SAFE_DIST)
+            while (yield rot_move.go(move_type='skid')).failure_reason is not '':
+               rot_move = rot_move.right(.25)
+               print rot_move
+
+            circle = navigator.move.d_circle_point(target_np, radius=ROT_SAFE_DIST, direction='ccw', theta_offset=-1.57)
+            for c in circle:
```
Comment (mattlangford) [inline mission_control/navigator_missions/nav_missions/circle_tower.py]: I'm going to try an lqrrt `drive!` circle here

File: mission_control/navigator_missions/nav_missions/circle_tower.py [hunk 3]
```diff
@@ -7,55 +7,114 @@
...
             fprint("No suitable totems found.", msg_color='red', title="CIRCLE_TOTEM")
-            continue
+            defer.returnValue(None)
+
+        target_np = navigator_tools.point_to_numpy(target.position)
+        set_up = navigator.move.look_at(target_np).set_position(target_np).backward(TOTEM_SAFE_DIST)
+        
+        # Approach totem, making sure we actually get there.
+        res = yield set_up.go(initial_plan_time=2)
+        while res.failure_reason is not '':
```
Comment (mattlangford) [inline mission_control/navigator_missions/nav_missions/circle_tower.py]: I should update the position of the totem here too

File: mission_control/navigator_missions/nav_missions/circle_tower.py [hunk 4]
```diff
@@ -7,55 +7,114 @@
...
 
-    while not all_found:
-        target_totem, explored_ids = yield get_closest_buoy(navigator, explored_ids)
+    # Get colors of intrest and directions
+    c1 = navigator.mission_params['totem_color_1'].get()
+    d1 = navigator.mission_params['totem_direction_1'].get()
+    c2 = navigator.mission_params['totem_color_2'].get()
+    d2 = navigator.mission_params['totem_direction_2'].get()
+    
+    colors = [c1, c2]
```
Comment (ironmig) [inline mission_control/navigator_missions/nav_missions/circle_tower.py]: The rules are bad, but I think there will be 3 to circle, and maybe 4 if they feel like it.

File: mission_control/navigator_missions/nav_missions/circle_tower.py [hunk 5]
```diff
@@ -7,55 +7,114 @@
...
+    
+    # TODO: What if we don't see the colors?
+    for color, direction in zip(colors, directions):
+        color = yield color
+        direction = yield direction
+
+        target = yield get_colored_buoy(navigator, color_map[color])
+        if target is None or _dist_from_bf(navigator_tools.point_to_numpy(target.position)) > (BF_WIDTH / 2 + BF_EST_COFIDENCE):
+            # Need to do something
             fprint("No suitable totems found.", msg_color='red', title="CIRCLE_TOTEM")
```
Comment (ironmig) [inline mission_control/navigator_missions/nav_missions/circle_tower.py]: Rules: ```Entering  the obstacle avoidance field and avoiding at least one   obstacle   or   circling   any   totem   will   be considered as an attempt at completing this task```

So we should definitely have a case to just circle a random totem and get out if we can't get the colors we want. Maybe even if there are no totems just drive to the buoy field and drive out and let rrt get us some points.

File: mission_control/navigator_missions/nav_missions/circle_tower.py (hunk 3 above)
Comment (ironmig) [inline mission_control/navigator_missions/nav_missions/circle_tower.py]: Agreed. Keep track of the ID of the totem for each colors totem and just query the database for the latest location of that ID.

File: .gitignore [hunk 6]
```diff
@@ -36,6 +36,7 @@ Thumbs.db
 *.pyc
 *.pyo
 *~
+*.swp
```
Comment (ironmig) [inline .gitignore]: legendary

**Comment by mattlangford** [comment]
I will address comments later this week and repush then

**Comment by ironmig** [comment]
Also either in this PR or a different one, please add the lakeday fixes to pose_editor.py, it fixed some bugs with discrete circles. 
https://github.com/uf-mil/Navigator/compare/dev...lake_day_11-20

**Comment by mattlangford** [comment]
I'll make a new one to add it to on Thursday night along with a couple other changes

**Comment by mattlangford** [comment]
Also I'm still tweaking params in the color stuff, but I'll push it on Thursday too

**Pull request merged**